src/components/OrderDetailWidget/OrderDetailWidget.tsx

```diff
@@ -1,73 +1,169 @@
-import React, { FC } from "react";
+import React, { FC, useMemo, useContext } from "react";
 import { useTranslation } from "react-i18next";
 import { useHistory } from "react-router-dom";
 
+import { FullOrder } from "@airswap/typescript";
 import { Web3Provider } from "@ethersproject/providers";
 import { useToggle } from "@react-hookz/web";
 import { useWeb3React } from "@web3-react/core";
+import { UnsupportedChainIdError } from "@web3-react/core";
 
-import { addDays } from "date-fns";
+import { BigNumber } from "bignumber.js";
+import compareAsc from "date-fns/compareAsc";
+import { ethers } from "ethers";
 
 import { nativeCurrencyAddress } from "../../constants/nativeCurrency";
+import { InterfaceContext } from "../../contexts/interface/Interface";
+import stringToSignificantDecimals from "../../helpers/stringToSignificantDecimals";
+import switchToEthereumChain from "../../helpers/switchToEthereumChain";
+import useInsufficientBalance from "../../hooks/useInsufficientBalance";
+import useTokenInfo from "../../hooks/useTokenInfo";
 import { OrderStatus } from "../../types/orderStatus";
 import { OrderType } from "../../types/orderTypes";
 import FeeModal from "../InformationModals/subcomponents/FeeModal/FeeModal";
+import { ButtonActions } from "../MakeWidget/subcomponents/ActionButtons/ActionButtons";
 import Overlay from "../Overlay/Overlay";
 import SwapInputs from "../SwapInputs/SwapInputs";
 import { Container, StyledInfoButtons } from "./OrderDetailWidget.styles";
 import ActionButtons from "./subcomponents/ActionButtons/ActionButtons";
 import OrderDetailWidgetHeader from "./subcomponents/OrderDetailWidgetHeader/OrderDetailWidgetHeader";
 
-// Dummy data, this data will come from store later.
-const expiry = addDays(new Date(), 1);
-const orderStatus = OrderStatus.open;
-const orderType = OrderType.private;
-const recipientAddress = nativeCurrencyAddress;
-const transactionLink =
-  "https://etherscan.io/tx/0x53ddb90435ab46d96035da71b4ff8a26bb0a682a7f2327aaeb8ff3848406204f";
+interface OrderDetailWidgetProps {
+  status: "idle" | "not-found" | "open" | "taken" | "canceled";
+  order: FullOrder;
+}
 
-const OrderDetailWidget: FC = () => {
+const OrderDetailWidget: FC<OrderDetailWidgetProps> = ({ status, order }) => {
   const { t } = useTranslation();
   const history = useHistory();
-  const { account } = useWeb3React<Web3Provider>();
-
+  const { setShowWalletList } = useContext(InterfaceContext);
+  const { active, account, error: web3Error } = useWeb3React<Web3Provider>();
   const [showFeeInfo, toggleShowFeeInfo] = useToggle(false);
+  const senderToken = useTokenInfo(order.senderToken);
+  const signerToken = useTokenInfo(order.signerToken);
+  const hasInsufficientTokenBalance = useInsufficientBalance(
+    signerToken,
+    ethers.utils.formatUnits(order.signerAmount, signerToken?.decimals!)
+  );
+
+  const orderStatus = () => {
```

Comment:
```suggestion
  const getOrderStatus = () => {
```

Put in OrderDetailWidget/helpers/index.ts

Reply:
_Put in OrderDetailWidget/helpers/index.ts_

Reply:
pushed